src/database/__tests__/firestore.spec.ts

```diff
@@ -109,9 +79,13 @@ describe('firestore', () => {
 
     describe('googleSignIn()', () => {
         it('uses GoogleAuthProvider for the redirect', () => {
-            firestore.googleSignIn().subscribe();
+            auth.signInWithRedirect.mockReturnValue(Promise.resolve());
 
-            expect(auth.signInWithRedirect).toHaveBeenCalledWith(new firebaseAuth.GoogleAuthProvider());
+            firestore.googleSignIn().subscribe(
+                () => {
+                    expect(auth.signInWithRedirect).toHaveBeenCalledWith(new firebaseAuth.GoogleAuthProvider());
```

Comment:
this test gives a false positive because you don't call the 'done()' callback here. To see what I mean, replace the expect statement with expect(true).toBe(false). You will see that it never fails the test unless done() is called immediately after.